### PR TITLE
Admin agency partner page navigation #562

### PR DESCRIPTION
**Pull request by pcaston**
Changed Agency Admin branches people and company navigation to use bootstrap pills navigation.
Harmonised with styles used in the job properties navigation.

### CODE REVIEW

**Comment by pcaston** [comment]
Re-running tests

**Comment by pcaston** [comment]
Re-open to rerun tests

File: features/step_definitions/web_steps.rb [hunk 1]
```diff
@@ -256,3 +256,7 @@ def search_text text
 Then(/^I should see "([^"]+)" in the email field$/) do |value|
   step %{The field 'Email' should have the value '#{value}'}
 end
+
+Then(/^I save the page as "([^"]+)"$/) do |screen|
+  page.save_screenshot("#{screen}", full:true)
```
Comment (pcaston) [inline features/step_definitions/web_steps.rb]: fixed

File: app/views/companies/_companies.html.haml [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
+        %td
+          - if company.pending_registration?
+            = link_to company.name, company_registration_path(company)
+          - else
+            = link_to company.name, company_path(company)
+        %td
+          = single_line_address(company.addresses.first)
+        %td
+          = status_desc(company)
+= will_paginate @companies, param_name: 'companies_page', |
```
Comment (pcaston) [inline app/views/companies/_companies.html.haml]: fixed

File: app/views/branches/_branches.html.haml [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
+            = branch.address.city
+          %td
+            = branch.address.street
+          %td
+            = branch.address.zipcode
+        - else
+          %td
+          %td
+          %td
+= will_paginate @branches, param_name: 'branches_page', |
```
Comment (pcaston) [inline app/views/branches/_branches.html.haml]: fixed

File: app/views/agency_admin/job_properties.html.haml [hunk 4]
```diff
@@ -6,16 +6,14 @@
...
-    %ul.nav.nav-pills.custom-pills{ role: 'tablist' }
-      %li.active{ role: 'presentation' }
-        %a{ href: '#job-specialties-pane',  'aria-controls' => 'job-specialties', role: 'tab', data: { toggle: 'tab' } }
-          Job Specialties
-        %li{ role: 'presentation' }
-          %a{ href: '#job-skills-pane',  'aria-controls' => 'job-skills', role: 'tab', data: { toggle: 'tab' } }
-            Job Skills
+    %ul.nav.nav-pills
+      %li.active
+        %a{data: {toggle: 'pill'}, href: '#job-specialties-pane'}Job Specialties
```
Comment (pcaston) [inline app/views/agency_admin/job_properties.html.haml]: fixed

File: app/views/agency_admin/job_properties.html.haml [hunk 5]
```diff
@@ -6,16 +6,14 @@
...
+      %li
+        %a{data: {toggle: 'pill'}, href: '#job-skills-pane'}Job Skills
```
Comment (pcaston) [inline app/views/agency_admin/job_properties.html.haml]: fixed

File: features/step_definitions/company_steps.rb [hunk 6]
```diff
@@ -1,6 +1,6 @@
 
 And /^I do not have an address$/ do
-	expect(page.find('#company_person_address_id')).not_to have_css("selected")
+	expect(page.find("#company_person_address_id")).not_to have_css("selected")
```
Comment (pcaston) [inline features/step_definitions/company_steps.rb]: fixed

**Comment by pcaston** [comment]
Re-run tests after fixes

**Comment by pcaston** [comment]
Re-open to re-run tests

**Comment by pcaston** [comment]
Fixed email validations

**Comment by patmbolger** [comment]
@pcaston - I worked on your PR to address the persistent test failures - see my PR starting with "TEST".  Changes that I made to your PR code:

1. agency_admin.feature
  - changed to driver `selenium`
  - took out wait steps (if followed by "I should see ...."
  - added a wait:
```
And I click the "Update Skill" button
And I wait 2 seconds    << added
Then "Update Job Skill" should not be visible
```

2. step_definitions/web_steps.rb - changed this method:
```
Then(/^(?:I|they) should( not)? see "([^"]*)"$/) do |not_see, string|
  unless not_see
    assert_text(string)   << changed
  else
    assert_no_text(string)  << changed
  end
end
```

3. job_application.feature - added one line near end of scenario:
```
Then I am in Company Admin's browser
And I am on the Company Person 'carter@ymail.com' Home page
And I wait 1 second
And I click the "Unassigned Company Tasks" link   << added
And I should see "Review job application"
And I should see "Job: software developer"
```

4. job_application_by_jd.feature - added last line shown here:
```
Then I am in Company Admin's browser
And I should see "Job Seeker: John Seeker has applied to this job"
And I am on the Company Person 'carter@ymail.com' Home page
And I wait 1 second
And I click the "Unassigned Company Tasks" link  << added
```

5. people_invite.feature - removed this line:
```
And I save the page as "screenshot.png"
```

**Comment by patmbolger** [comment]
Here is an image for the first page:
screen shot 2017-03-16 at 5 38 52 am

This is from the job properties page:
screen shot 2017-03-16 at 5 40 48 am


1. The left-hand border of the first button ("Agency Information") should align with the the start of the page header "PETS Administration ...")

2. The buttons should be spaced below the header the same distance as for job properties page.

3. The header should not extend further to the left than the first main menu item ("MET|PLUS").  Ideally, it should be indented to the right a bit, as in the job properties page.

4. The "Edit Agency" button should be brought up and placed in the same position as the "Add job specialty" button in the job properties page.

5. The same comments as above apply to the similar buttons on the other tabs ("Add Branch", "Invite Person").

**Comment by pcaston** [comment]
Reopen pull request to re-run tests.

File: app/views/agency_admin/home.html.haml [hunk 7]
```diff
@@ -1,68 +1,55 @@
...
+
+    .tab-content
+      .tab-pane.fade.in.active#agency_table
+        .col-xs-12.text-right
+          = button_to 'Edit Agency', edit_agency_path(@agency), method: :get,
+            class: 'btn btn-primary btn-sm pull-right'
+        .col-xs-12.pad-0
+        = render partial: 'agencies/agency', object: @agency
+        .clearfix
+
```
Comment (pcaston) [inline app/views/agency_admin/home.html.haml]: Fixed

File: app/views/agency_admin/home.html.haml [hunk 8]
```diff
@@ -1,68 +1,55 @@
...
+          - else
+            = render 'branches/branches'
+        .clearfix
+
+      .tab-pane.fade#people_table
+        .col-xs-12.text-right
+          = link_to 'Invite Person',
+            new_user_invitation_path(person_type: 'AgencyPerson',
+            org_id: @agency.id),
+            method: :get, class: 'btn btn-primary btn-sm pull-right'
```
Comment (pcaston) [inline app/views/agency_admin/home.html.haml]: fixed

**Comment by patmbolger** [comment]
@pcaston - Looking much better Paul.  I did a little styling myself in order to better align the the two agency admin pages  (yours and Arun's) .  The files that I touched (relative to your PR) were:
```
app/assets/stylesheets/application.css
app/views/agencies/_agency.html.haml
app/views/agency_admin/home.html.haml
app/views/agency_admin/job_properties.html.haml
app/views/agency_people/_agency_people.html.haml
app/views/branches/_branches.html.haml
app/views/companies/_companies.html.haml
```
Note that I also made corresponding changes to the job_properties page.

With these changes, this is what the admin pages look like now:

---

screen shot 2017-03-22 at 8 21 25 am

---

screen shot 2017-03-22 at 8 21 47 am

---

screen shot 2017-03-22 at 8 21 55 am

---

screen shot 2017-03-22 at 8 22 03 am

---

screen shot 2017-03-22 at 8 36 46 am


---

screen shot 2017-03-22 at 8 36 52 am

**Comment by pcaston** [comment]
Repopen to re-run tests

**Pull request merged**